### PR TITLE
Add Github Actions Example

### PR DESCRIPTION
**Pull request by Speediing**
### Description

Added a basic example of how to use the Vercel CLI to deploy using GitHub Actions for users unable to use the automatic integration (ex. GitHub Enterprise Server). Added a new directory because it didn't seem to fit elsewhere and we will most likely want more examples for other providers (Azure, Gitlab etc.), but feel free to direct me to the correct place.

### Demo URL

N/A

### Type of Change

- [x] New Example
- [ ] Example updates (Bug fixes, new features, etc.)
- [ ] Other (changes to the codebase, but not to examples)

### New Example Checklist

- [x] 🛫 `npm run new-example` was used to create the example
- [ ] 📚 The template wasn't used but I carefuly read the [Adding a new example](https://github.com/vercel/examples#adding-a-new-example) steps and implemented them in the example
- [ ] 📱 Is it responsive? Are mobile and tablets considered?

### CODE REVIEW

**Review by lfades — CHANGES_REQUESTED** [review]
@Speediing Hi there and thank you so much for this PR!

What do you think about renaming it to `/solutions/vercel-cli-and-github-actions`? It's more descriptive and that's what the solutions folder is for!

We would also need to setup a demo deployment for it as we do for the other examples, I guess that'll imply adding a very similar github action at the root of the repository, I'm not sure if it's possible to use the one provided in the example as it is.

**Comment by leerob** [comment]
We're wanting to add a `ci-cd` folder to start to add a bunch of examples for other providers 😄

File: ci-cd/github-actions/README.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+This will create a `.vercel` folder in your project where you will find a `project.json` file with both your `"projectId"` and `"orgId"`. Note these for later because we will need them to link from Github.
+
+Finally make sure you have created a [Vercel API token](https://vercel.com/support/articles/how-do-i-use-a-vercel-api-access-token) to give Github the right authentication to push to your Vercel account.
+
+Now go to your Github repository and add as [secrets](https://docs.github.com/en/actions/security-guides/encrypted-secrets) the `VERCEL_ORG_ID`, `VERCEL_PROJECT_ID` and `VERCEL_TOKEN` we created previously.
+
+This finishes off all the configuration neccesary and now you can try out the workflow:
+
+- Create a new PR in Github
```
Comment (leerob) [inline ci-cd/github-actions/README.md:80]: Thoughts on making the "application" we deploy just an HTMl file? Would love to simplify things here.

Comment (leerob) [reply ci-cd/github-actions/README.md:80]: https://github.com/leerob/actions-cd-cd/pull/1

**Comment by Speediing** [comment]
> We would also need to setup a demo deployment for it as we do for the other examples, I guess that'll imply adding a very similar github action at the root of the repository, I'm not sure if it's possible to use the one provided in the example as it is.

I'm thinking for the demo deploy I can set it up with an automatic vercel deployment and add the link in the readme? Otherwise it would start getting complex running a separate GH action at the root, especially as we add examples for other CICD providers. I'm good with whatever you think though @lfades .

File: ci-cd/github-actions/README.md (hunk 1 above)
Comment (Speediing) [reply ci-cd/github-actions/README.md:80]: Sounds perfect to me, the main thing here is just the ci yaml files 🚀

File: ci-cd/github-actions/README.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
+---
+name: Github Actions Deploy
+slug: github-actions-deploy
+description: Learn how to use the Vercel CLI and Github actions to deploy to Vercel with your own CICD
```
Comment (leerob) [inline ci-cd/github-actions/README.md:4]:
```suggestion
description: Deploy to Vercel with your customer GitHub Actions CI/CD.
```

File: ci-cd/github-actions/README.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+framework: Next.js
```
Comment (leerob) [inline ci-cd/github-actions/README.md:5]:
```suggestion
```

File: ci-cd/github-actions/README.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+---
+
+# Bring Your Own CI/CD - Github Actions
+
+This example shows how to use the Vercel CLI and Github Actions to deploy to Vercel with your own custom pipeline. You can generate preview deployments, deploy to production when code is merged into the main branch, and enable rollbacks by reverting PRs. This is useful for both users of Github.com who want more CI/CD control as well as Github Enterprise Server users who can't leverge the built-in integration.
```
Comment (leerob) [inline ci-cd/github-actions/README.md:10]:
```suggestion
This example shows how to use the Vercel CLI and Github Actions to deploy to Vercel with your own custom pipeline. You can generate preview deployments, deploy to production when code is merged into the main branch, and enable rollbacks by reverting PRs. This is useful for both users of Github.com who want more CI/CD control as well as Github Enterprise Server users who can't leverage Vercel's built-in git integration.
```

File: ci-cd/github-actions/README.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+## How to Use
+
+### Project Setup
+
+When you clone the repo you will notice there are two example pipelines for a trunk-based development workflow in the `.github/workflows` folder. One is for preview and and one is for production deployments:
```
Comment (leerob) [inline ci-cd/github-actions/README.md:16]:
```suggestion
After you clone the repo, you will notice there are two example pipelines for a trunk-based development workflow in the `.github/workflows` folder. One is for preview and and one is for production deployments:
```

File: ci-cd/github-actions/README.md [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+        run: npm install --global vercel@canary
+      - name: Pull Vercel Environment Information
+        run: vercel pull --yes --environment=preview --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Build Project Artifacts
+        run: vercel build --token=${{ secrets.VERCEL_TOKEN }}
+      - name: Deploy Project Artifacts to Vercel
+        run: vercel deploy --prebuilt --token=${{ secrets.VERCEL_TOKEN }}
+```
+
+You will see that we need to setup three secrets (VERCEL_ORG_ID, VERCEL_PROJECT_ID, VERCEL_TOKEN) in this action, these will be covered in the next section.
```
Comment (leerob) [inline ci-cd/github-actions/README.md:42]:
```suggestion
You will see that we need to setup three secrets (`VERCEL_ORG_ID`, `VERCEL_PROJECT_ID`, `VERCEL_TOKEN`) in this action, these will be covered in the next section.
```

File: ci-cd/github-actions/README.md (hunk 2 above)
Comment (Speediing) [reply ci-cd/github-actions/README.md]: Appreciate the feedback! 🐐

**Review by leerob — APPROVED** [review]

**Pull request merged**